provider_test.go

```diff
@@ -0,0 +1,43 @@
+package main
```

Comment:
I'd like if we can move the provider_test.go file under the tests folder.

Reply:
I've moved the files into appropriate packages now